File: mic-mydocs/src/main/java/dev/skypea/mic_mydocs/workspace/domain/model/Workspace.java
package dev.skypea.mic_mydocs.workspace.domain.model;

import lombok.Builder;
import lombok.Value;

import java.time.OffsetDateTime;
import java.util.List;

@Value
@Builder
public class Workspace {
    String id;
    String name;
    String ownerId;
    OffsetDateTime createdAt;
    List<WorkspaceMember> members;
}

File: spa-mydocs/src/features/workspaces/domain/model/Workspace.ts
export interface WorkspaceMember {
  userId: string;
  nickname: string;
  email: string;
  picture: string | null;
}

export interface Workspace {
  id: string;
  name: string;
  ownerId: string;
  createdAt: string;
  members: WorkspaceMember[];
  role: 'OWNER' | 'MEMBER';
}
